_posts/2022-04-08-reinforcement-learning.md

```diff
@@ -0,0 +1,117 @@
+
+The racing part is also a bit addictive, due to competitiveness. There are constantly virtual races going on, pretty easy to attend with your own models. If you climb to the top 10 in the open track results (quite a feat by the way), you can enter the pro races, with more challenging tracks and setups. And best of the pro racers then move on to AWS re:Invent for the really big races.
+
+## Why is it useful?
+
+Idea was to pick something that would unlikely be part of our customer projects, or customer needs, just for fun. It leaves more artistic freedom to play with it, as you like, simply focus on the fun of learning.
+
+That being said.. There's a huge demand for people who understand the cloud services well, and even though Deepracer is just a toy, it's still a great way to learn about the AWS services. In fact, it's essential to learn at least a bit of them. While services like IAM, S3, and Sagemaker have everyday uses, the Robomaker environment is a bit more exotic, which might be fun to learn as well.
+
+Additionally - who's to say there would not be customer demand in this area, too? That might be a happy accident. Perhaps someone wants to train an autonomous bot, car, or drone to deliver food, purchases, and supplies. Or challenge Tesla at their own game. Or build automated systems to pick up people's shopping carts from shelves, and put them in delivery boxes. The future will bring more autonomous systems, and people who are experienced enough to be able to make them robust and safe are always in demand. And this is a great way to start that path.
+
+## What have I learned so far from this?
+
+I used to race with Deepracer some years back, then life happened, and had a bit of a break there. Now that I decided to resurrect my old interest, it's been fun to see it's still much alive. Some new things have also happened: The local training has become more viable and easy to get started with. And the new obstacle avoidance mode has been added. Obviously, lots more tracks are available.
+
+Last time I was training I used a combination of an Ubuntu machine with a great graphics card, and sometimes the AWS console services. This time I was happy to see that I can also do local training with Windows, WSL2, and a Docker installation, and it works great also with the AWS console. So I can train a model locally, once it's good enough, I can refine it in the AWS console, and race with it. This is a very cost-effective way to have some fun.
+
+Other insights. As a coder, I've been very fixated on the reward algorithm. Now I'm also trying to understand the hyperparameters, and action space, a lot better. Some experiments I've been running:
+
+- Custom action space where the fastest speeds do not include dramatic turns: I noticed when you up the speed a lot, the car starts to spin due to simulation physics. I can customize allowed turn angles in the action space, allowing tighter turns only for the slower speeds.
```

Comment:
`I noticed when you up the speed a lot` -> `I noticed when you increase the speed a lot` ?